### DISCUSSION TITLE
What are direct commands?

### DISCUSSION THREAD

**Discussion by Zombie1995** [luc-github/ESP3D#599] [Questions]
How to use them? Just using serial port or by another way?

**Comment by luc-github** [comment]
you type then in console webUI , in serial console, in telnet, in macro files, in SD file processed by printer eventually using the serial output command of FW printer if necessary
So to sum up any way ESP3D can read the command

https://github.com/luc-github/ESP3D/wiki/Direct-ESP3D-commands

**Reply by Zombie1995** [reply]
Thank you very much!!!
